apis/rabbitmq/v1beta1/transporturl_types.go

```diff
@@ -0,0 +1,62 @@
+/*
+Copyright 2022.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1beta1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// EDIT THIS FILE!  THIS IS SCAFFOLDING FOR YOU TO OWN!
+// NOTE: json tags are required.  Any new fields you add must have json tags for the fields to be serialized.
+
+// TransportURLSpec defines the desired state of TransportURL
+type TransportURLSpec struct {
+	// +kubebuilder:validation:Required
+	// RabbitmqClusterName the name of the Rabbitmq cluster which to configure the transport URL
+	RabbitmqClusterName string `json:"rabbitmqClusterName,omitempty"`
+}
+
+// TransportURLStatus defines the observed state of TransportURL
+type TransportURLStatus struct {
+	// SecretName - name of the secret containing the rabbitmq transport URL
+	SecretName string `json:"secretName,omitempty"`
```

Comment:
I will add them

Reply:
@gibizer Andrew is working on adding conditions in general to the openstack-operator. Could we hold on adding them and do them once we have basic condition support in.

Reply:
Is TransportURL expected to be created by openstack-operator and not nova- and cinder-operator? If so then I'm OK to add condition to TransportURL when that condition is first used by the openstack-operator.

Reply:
It will end up being similar to MariadbDatabase. We'll instantiate the creation of the TransportURL CR's from cinder/nova/neutron. Will likely have a lib-common function to help with this however.

Reply:
I see. Thanks for the explanation. I'm OK adding the condition later.